### PR TITLE
ActionManager skeleton

### PR DESCRIPTION
**Pull request by kebeclibre**

### CODE REVIEW

**Comment by kebeclibre** [comment]
@aab-odoo 
@ged-odoo 
A skeleton, ultra wip, for actionManager as Service

File: addons/wowl/static/src/components/actionManager/actionManager.xml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<templates xml:space="preserve">
+
+    <div t-name="wowl.ActionManager" owl="1">
+        <NavBar/>
```
Comment (ged-odoo) [inline addons/wowl/static/src/components/actionManager/actionManager.xml]: why the navbar inside the action manager?

File: addons/wowl/static/src/components/actionManager/actionManager.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import { Component, hooks } from "@odoo/owl";
+import { useService } from "./../../services";
+import type { OdooEnv } from "./../../env";
+import { NavBar } from "../navbar/navbar";
+
+const { onMounted , onPatched } = hooks;
+
+export class ActionManager extends Component<{}, OdooEnv> {
```
Comment (ged-odoo) [inline addons/wowl/static/src/components/actionManager/actionManager.ts]: maybe rename it to `ActionContainer`

File: addons/wowl/static/src/services/action_manager.ts [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+      switch (property) {
+        case "slots":
+          return [main];
+      }
+    },
+    dispatch
+  };
+}
+export const actionManagerService: Service<ActionManager> = {
+  name: "actionManager",
```
Comment (ged-odoo) [inline addons/wowl/static/src/services/action_manager.ts]: rename service to `action_manager`

File: addons/wowl/static/src/components/actionManager/actionManager.xml (hunk 1 above)
Comment (kebeclibre) [reply addons/wowl/static/src/components/actionManager/actionManager.xml]: Most of its part are managed by the ActionManager: current menu (linked to the current action).
I am not sure of that design though...

File: addons/wowl/static/src/services/action_manager.ts (hunk 3 above)
Comment (ged-odoo) [reply addons/wowl/static/src/services/action_manager.ts]: i expect the action manager service to depends on:
- `router` service => to read/update the url
- `rpc` service => to perform rpc (such as load action)
- `menus` service => to query menu info if necessary (not sure about this one)

File: addons/wowl/static/src/services/action_manager.ts [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+  do(cmd: string, args: any): any
+  get(property: string, args: any): any
+}
+
+class GenericAction extends Component<{}, OdooEnv> {
+  static template = tags.xml`
+  <div>
+    <p t-esc="controller.display" />
+    <p>all</p>
+    <t t-foreach="getAllControllers()" t-as="ct">
```
Comment (ged-odoo) [inline addons/wowl/static/src/services/action_manager.ts]: looks overengineered

File: addons/wowl/static/src/services/action_manager.ts [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+}
+
+//type finalizeParams = 'commit' | 'rollback';
+
+interface ActionManager {
+  get(property: string): any,
+  dispatch(method: string, params: dispatchParams): void,
+}
+
+interface Slot {
```
Comment (ged-odoo) [inline addons/wowl/static/src/services/action_manager.ts]: probably bad name... slot is used in owl. not sure about this

File: addons/wowl/static/src/components/actionManager/actionManager.xml (hunk 1 above)
Comment (ged-odoo) [reply addons/wowl/static/src/components/actionManager/actionManager.xml]: i disagree. the navbar is just connected to the menu service. that's it

File: addons/wowl/static/src/components/navbar/navbar.xml [hunk 6]
```diff
@@ -21,7 +21,7 @@
                     t-att-data-menu-id="app.id"
                     t-att-data-menu-xmlid="app.xmlid"
                     t-att-data-action-id="app.actionID"
-                    t-on-click.prevent="trigger('open-menu', { menuID: app.id })">
+                    t-on-click.prevent="am.dispatch('doAction', { display: app.id })">
```
Comment (ged-odoo) [inline addons/wowl/static/src/components/navbar/navbar.xml]: boom, headshot

**Review by ged-odoo — APPROVED** [review]

**Pull request merged**